changelog/unreleased/issue-2571

```diff
@@ -0,0 +1,14 @@
+Enhancement: Check contents of unchanged files
+
+We've added an extra check whether the contents are available in the index when backing
+up unchanged files.
+
+Backup used to not save the contents of unchanged files at all - even if their contents
+were not available, maybe due to disc corruption or accidential removal.
```

Comment:
disk, accidental

Reply:
These two causes are probably only a small fraction of all causes of damaged repositories that have appeared in the issues.